docs/proposals/promote-dependencies-automatically/README.md

```diff
@@ -0,0 +1,196 @@
+
+### Example
+
+Assuming we deploy a workload CR named `myapp` and its referenced ConfigMap called `my-config` in the member cluster member1.
+
+```yaml
+apiVersion: apps/v1
+kind: Workload
+metadata:
+  name: myapp
+  labels:
+    app: myapp
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: myapp
+  template:
+    metadata:
+      labels:
+        app: myapp
+    spec:
+      containers:
+        - image: nginx
+          name: nginx
+          volumeMounts:
+            - name: configmap
+              mountPath: "/configmap"
+      volumes:
+        - name: configmap
+          configMap:
+            name: my-config
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+data:
+  nginx.properties: |
+    proxy-connect-timeout: "10s"
+    proxy-read-timeout: "10s"
+    client-max-body-size: "2m"
+```
+After executing the `karmadactl promote workload myapp -c member1 -d=true` command, we can obtain the Resource Binding and Work object corresponding to the workload `myapp` and ConfigMap `my-config` in the Karmada control plane. This indicates that both `myapp` and its dependent resource `my-config` have been successfully promoted to the Karmada control plane.
+
+When deleting `myapp` from the Karmada control plane, `myapp` in the member cluster member1 will also be deleted.
+
+### Test Plan
+
+#### UT
+
+Testing coverage for new methods.
+
+#### E2E
+
+| Test Case                                                 | Case Description                                                                                                                |
+|-----------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------|
+| Promote dependencies of native resource automatically     | Test if dependencies of native resource automatically promote to karmada.                                                       |
+| Promote dependencies of thirdparty resource automatically | Test if dependencies of thirdparty resource automatically promote to karmada.                                                   |
+| Promote dependencies of custom resource automatically     | Test if dependencies of custom resource automatically promote to karmada(use lua configuration way and webhook interpret way).  |
+
+
+## Alternatives
+### TODO: Port forward the request to webhook interpret service in the cluster
+
+Generally, external access to applications (Pods) running in Kubernetes can be achieved through three modes: NodePorts, LoadBalancer, and Ingress. Additionally, the `kubectl` subcommand `port-forward` can be used to forward ports and map local ports to specific Pod ports, allowing access to Pod services within the Kubernetes cluster.
```

Comment:
I think this chapter should be moved to the front. Alternatives means the other implementation way

Reply:
OK